File: codesignal/challenges/longestPath/longestPath.py
def longestPath(fileSystem):
    s = []
    curlen = 0
    maxlen = 0
    fileSystem = fileSystem.splitlines()
    for f in fileSystem:
        name = f.lstrip('\t')
        # level
        l = (len(f) - len(name))
        if s:
            for _ in range(len(s)-l):
                s.pop()

        s.append(name)
        curlen = sum([len(x) for x in s]) + len(s)-1
        if '.' in name and curlen > maxlen:
            maxlen = curlen

    return maxlen

File: codesignal/challenges/longestPath/longestPath.rs
fn longestPath(fileSystem: String) -> i32 {
    let mut stack: Vec<String> = Vec::new();
    let lines: Vec<&str> = fileSystem.split('\u{C}').collect();

    let mut curlen = 0;
    for name in lines {
        let fileName = name.replace('\t', "");
        let level = name.len() - fileName.len();

        if stack.len() > 0 {
            while level < stack.len() {
                stack.pop();
            }
        }
        stack.push((*fileName).to_string());

        let mut len = stack.len() - 1;
        for s in &stack {
            len += s.len();
        }

        if fileName.contains('.') && len > curlen {
            curlen = len
        }
    }

    return curlen as i32;
}

